test/e2e/util/util.go

```diff
@@ -1378,9 +1383,114 @@ func ValidateMetrics(t *testing.T, ctx context.Context, hc *hyperv1.HostedCluste
 	})
 }
 
+func createIngressRoute53Record(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster, clusterOpts *PlatformAgnosticOptions) {
+	t.Helper()
+	g := NewWithT(t)
+	t.Logf("Creating Ingress Route53 Record for HostedCluster %s", hostedCluster.Name)
+
+	guestClient := WaitForGuestClient(t, ctx, client, hostedCluster)
+	defaultIngressRouterService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "router-default",
+			Namespace: "openshift-ingress",
+		},
+	}
+
+	err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 30*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+		err = guestClient.Get(ctx, crclient.ObjectKeyFromObject(defaultIngressRouterService), defaultIngressRouterService)
+		if err != nil {
+			return false, nil
```

Comment:
shouldn't an error be returned here?

Reply:
no, it returns "done == false", but no error. In fact in this function we never return an error. It seems to be a pattern that has been used elsewhere around here so I'll keep it unless this is problematic.

Reply:
I addressed this comment and now we return an error.